### PR TITLE
Add config.readyEvent and config.delay

### LINKED ISSUE #123: Asynchronous injected JS doesn't get to finish
@ihodes ran into this, if you inject some JS along the lines of:

``` javascript
window.setTimeout(function() {
  // click something
}, 1000);
```

Then the screenshot gets taken before the click happens. The [page.waitForReady](https://github.com/bslatkin/dpxdt/blob/db3818aecf998a70982fcc54227a0bfe14faaa43/dpxdt/client/capture.js#L283) function waits for XHRs to complete, but it doesn't wait for other async code to run. I'm not even sure if this is possible with Phantom.

### PR DESCRIPTION
**Pull request by gentoo90**
If `config.readyEvent` is set, `phantomjs` will wait for exactly this `console.log` message
before taking a screenshot.

`config.delay` is applied before taling a screenshot (after `readyEvent` if it's set).

So with config
```json
{
  ...
  "readyEvent": "captureScreen",
  "delay": 1000
}
```
`phantomjs` will wait for one second after the string `captureScreen` is logged to the console.

Both features are from [BackstopJS](https://github.com/garris/BackstopJS#trigger-screen-capture-via-consolelog).

Fixes #123.

### CODE REVIEW

**Comment by bslatkin** [comment]
Sorry for my delay. Thanks for taking the time to put this together. The Travis build is broken so I think this patch probably has a bug in it?

Why do you need the `page.readyEventOccured = true;` part of the patch? I don't understand why using `console.log` is the best approach. Are you just trying to get around the security boundary?

What I'd do instead is having `doInject` do a `page.evaluate()` call to see if an attribute on `window` is set. If it's not, then go back to waitForReady until the attribute is done. You can define a callback function before the page rendering using `page.evaluate()` as well. Then use that instead of the `console.log` you're relying on.

Basically, I'm asking for you to add an API like `window.dpxdtReady()` within the page and a config option that says to wait for that function. Does that make sense?

**Comment by gentoo90** [comment]
I launched travis build again on the same commit and [tests succeeded](https://travis-ci.org/gentoo90/dpxdt/builds/114378616). The test in pull request failed with
```
self.assertEquals(work_queue.WorkQueue.DONE, found.status)
AssertionError: 'done' != u'live'
```
so probably some task from the queue didn't finish in time.

I used `console.log` to trigger `readyEvent` because pages may already have some log messages like `page loaded` and this approach will allow to test them without modifying their code or injecting anything.

**Comment by gentoo90** [comment]
I added `window.dpxdtReady` field as a possible trigger for page ready event.
But I still would like to be able to use `console.log()`, from pages own code for example.
@bslatkin could you merge both of this features?
